### PR TITLE
feat: service layer abstraction

### PR DESCRIPTION
**Pull request by cofin**
feat: adds a service layer for the repositories that can be extended

### CODE REVIEW

File: advanced_alchemy/service/_async.py [hunk 1]
```diff
@@ -0,0 +1,526 @@
...
+
+        Returns:
+            Representation of created instance.
+        """
+        data = await self.to_model(data, "create")
+        return await self.repository.add(data)
+
+    async def create_many(
+        self,
+        data: list[ModelT | dict[str, Any]] | list[dict[str, Any]] | list[ModelT],
```
Comment (Alc-Alc) [inline advanced_alchemy/service/_async.py:94]: Can this and the other occurrences be replaced with `ModelDictListT` alias defined in `advanced_alchemy/service/typing.py`?

```suggestion
        data: ModelDictListT
```

**Review by Alc-Alc — APPROVED** [review]

File: advanced_alchemy/service/_async.py (hunk 1 above)
Comment (cofin) [reply advanced_alchemy/service/_async.py:94]: It ended up complaining in a few other places on the repository then.  I'm sure there's a way to clean these up though.

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -45,6 +45,19 @@ jobs:
 
       - name: Install Microsoft ODBC Drivers
         run: sudo ACCEPT_EULA=Y apt-get install msodbcsql18 -y
+      - name: Free additional space
```
Comment (provinzkraut) [inline .github/workflows/ci.yaml:48]: Just curious, why is this needed?

**Review by provinzkraut — APPROVED** [review]

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (cofin) [reply .github/workflows/ci.yaml:48]: We've had quite a few failures because we go beyond the 20gb or so that Github allocates

Comment (provinzkraut) [reply .github/workflows/ci.yaml:48]: Oh :eyes:
